lang/semantic_token.go

```diff
@@ -30,7 +30,7 @@ const (
 	TokenObjectKey     SemanticTokenType = "hcl-objectKey"
 	TokenMapKey        SemanticTokenType = "hcl-mapKey"
 	TokenKeyword       SemanticTokenType = "hcl-keyword"
-	TokenTraversalStep SemanticTokenType = "hcl-traversalStep"
+	TokenReferenceStep SemanticTokenType = "hcl-referenceStep"
```

Comment:
I think we'll also need to reflect this in both LS (incl. docs) and vscode-terraform (incl. docs).